keyboards/ploopyco/ploopyco.c

```diff
@@ -16,118 +16,139 @@
+pin_t    encoder_pins_b[1] = ENCODERS_PAD_B;
 bool     debug_encoder     = false;
-bool     is_drag_scroll    = false;
 
 bool encoder_update_kb(uint8_t index, bool clockwise) {
     if (!encoder_update_user(index, clockwise)) {
         return false;
     }
-#ifdef MOUSEKEY_ENABLE
+#    ifdef MOUSEKEY_ENABLE
     tap_code(clockwise ? KC_WH_U : KC_WH_D);
-#else
+#    else
     report_mouse_t mouse_report = pointing_device_get_report();
-    mouse_report.v = clockwise ? 1 : -1;
+    mouse_report.v              = clockwise ? 1 : -1;
     pointing_device_set_report(mouse_report);
     pointing_device_send();
-#endif
+#    endif
     return true;
 }
 
 void encoder_driver_init(void) {
-    setPinInput(OPT_ENC1);
-    setPinInput(OPT_ENC2);
-
+    for (uint8_t i = 0; i < ARRAY_SIZE(encoder_pins_a); i++) {
+        gpio_set_pin_input(encoder_pins_a[i]);
+        gpio_set_pin_input(encoder_pins_b[i]);
+    }
     opt_encoder_init();
 }
 
 void encoder_driver_task(void) {
-    // Lovingly ripped from the Ploopy Source
+    uint16_t p1 = analogReadPin(encoder_pins_a[0]);
+    uint16_t p2 = analogReadPin(encoder_pins_b[0]);
 
+    if (debug_encoder) dprintf("OPT1: %d, OPT2: %d\n", p1, p2);
+
+    int8_t dir = opt_encoder_handler(p1, p2);
     // If the mouse wheel was just released, do not scroll.
-    if (timer_elapsed(lastMidClick) < SCROLL_BUTT_DEBOUNCE) {
+    if (timer_elapsed(lastMidClick) < PLOOPY_SCROLL_BUTTON_DEBOUNCE) {
         return;
     }
 
     // Limit the number of scrolls per unit time.
-    if (timer_elapsed(lastScroll) < OPT_DEBOUNCE) {
+    if (timer_elapsed(lastScroll) < PLOOPY_SCROLL_DEBOUNCE) {
         return;
     }
 
     // Don't scroll if the middle button is depressed.
     if (is_scroll_clicked) {
-#ifndef IGNORE_SCROLL_CLICK
+#    ifndef PLOOPY_IGNORE_SCROLL_CLICK
         return;
-#endif
+#    endif
     }
 
-    lastScroll  = timer_read();
-    uint16_t p1 = adc_read(OPT_ENC1_MUX);
-    uint16_t p2 = adc_read(OPT_ENC2_MUX);
-    if (debug_encoder) dprintf("OPT1: %d, OPT2: %d\n", p1, p2);
-
-    int dir = opt_encoder_handler(p1, p2);
-
     if (dir == 0) return;
-    encoder_queue_event(0, dir == 1);
+    encoder_queue_event(0, dir > 0);
+    lastScroll = timer_read();
 }
+#endif
 
 report_mouse_t pointing_device_task_kb(report_mouse_t mouse_report) {
     if (is_drag_scroll) {
-        mouse_report.h = mouse_report.x;
+        scroll_accumulated_h += (float)mouse_report.x / PLOOPY_DRAGSCROLL_DIVISOR_H;
+        scroll_accumulated_v += (float)mouse_report.y / PLOOPY_DRAGSCROLL_DIVISOR_V;
+
+        // Assign integer parts of accumulated scroll values to the mouse report
+        mouse_report.h = (int8_t)scroll_accumulated_h;
 #ifdef PLOOPY_DRAGSCROLL_INVERT
-        // Invert vertical scroll direction
-        mouse_report.v = -mouse_report.y;
+        mouse_report.v = -(int8_t)scroll_accumulated_v;
 #else
-        mouse_report.v = mouse_report.y;
+        mouse_report.v = -(int8_t)scroll_accumulated_v;
 #endif
```

Comment:
`-(int8_t)scroll_accumulated_v;` is repeated twice here. 

Trying this branch `PLOOPY_DRAGSCROLL_INVERT` has no effect due to this change. 

Removing the negative modifier for the `PLOOPY_DRAGSCROLL_INVERT` ifdef branch seems to have the desired effect for me.

Reply:
This is perhaps personal preference, but I may expect both h and v values here to be inverted. It's unclear from the documentation if the `PLOOPY_DRAGSCROLL_INVERT` is only supposed to be for vertical scrolling. The reason I choose invert is to make the trackball scroll behavior to behave closer to a touch device when "moving" the plane around. 

```c
#ifdef PLOOPY_DRAGSCROLL_INVERT
        mouse_report.h = -(int8_t)scroll_accumulated_h;
        mouse_report.v = (int8_t)scroll_accumulated_v;
#else
        mouse_report.h = (int8_t)scroll_accumulated_h;
        mouse_report.v = -(int8_t)scroll_accumulated_v;
#endif
```

Reply:
Other than the unneeded negative for the non-inverted, I'd rather keep it as is, as that is closer to the older behavior.

Reply:
Yup understood it was like that before, I only left the second comment because I still wasn't able to get the "invert" behavior to be quite right. 

Please feel free to leave it was it was!